tket/src/WeightSubgrMono/Reducing/CloseVerticesFilter.cpp

```diff
@@ -0,0 +1,124 @@
+// Copyright 2019-2022 Cambridge Quantum Computing
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "WeightSubgrMono/Reducing/CloseVerticesFilter.hpp"
+
+#include <cmath>
+#include <sstream>
+
+#include "WeightSubgrMono/Searching/FixedData.hpp"
+#include "WeightSubgrMono/Searching/SearchNodeWrapper.hpp"
+
+namespace tket {
+namespace WeightedSubgraphMonomorphism {
+
+// In experiments, this appeared to be the best value
+// (in combination with all other reductions).
+CloseVerticesFilter::CloseVerticesFilter() : m_num_levels(2) {}
```

Comment:
It's this value because I did a lot of experiments to find the best value. We probably don't want to allow the user to change it.

But yeah, I suppose, let them set it and hang themselves if they really insist.

Reply:
I'm not against it being hardcoded - but if it is maybe just have it set in the class and not at construction? With some explanation as to how this value was reached in the comments

Reply:
Yes, this is now in MainSolverParameters.